src/domains/citizen-auth-app/04_function_lollipop.tf

```diff
@@ -44,26 +45,88 @@ locals {
       FIRST_LC_ASSERTION_CLIENT_BASE_URL         = "https://api.io.pagopa.it"
       FIRST_LC_ASSERTION_CLIENT_SUBSCRIPTION_KEY = data.azurerm_key_vault_secret.first_lollipop_consumer_subscription_key.value
     }
+
+    # Scaling strategy
+    # 05 - 19,30 -> min 3
+    # 19,30 - 23 -> min 4
+    # 23 - 05 -> min 2
+    autoscale_profiles = [{
+      name = "{\"name\":\"default\",\"for\":\"evening\"}",
+
+      recurrence = {
+        hours   = 22
+        minutes = 59
+      }
+
+      capacity = {
+        default = var.function_lollipop_autoscale_default + 1
+        minimum = var.function_lollipop_autoscale_minimum + 1
+        maximum = var.function_lollipop_autoscale_maximum
+      }
+      },
+      {
+        name = "{\"name\":\"default\",\"for\":\"night\"}",
+
+        recurrence = {
+          hours   = 5
+          minutes = 0
+        }
+
+        capacity = {
+          default = var.function_lollipop_autoscale_default + 1
+          minimum = var.function_lollipop_autoscale_minimum + 1
+          maximum = var.function_lollipop_autoscale_maximum
+        }
+      },
+      {
+        name = "evening",
+
+        recurrence = {
+          hours   = 19
+          minutes = 30
+        }
+
+        capacity = {
+          default = var.function_lollipop_autoscale_default + 2
+          minimum = var.function_lollipop_autoscale_minimum + 2
+          maximum = var.function_lollipop_autoscale_maximum
+        }
+      },
+      {
+        name = "night",
+
+        recurrence = {
+          hours   = 23
+          minutes = 0
+        }
+
+        capacity = {
+          default = var.function_lollipop_autoscale_default
+          minimum = var.function_lollipop_autoscale_minimum
+          maximum = var.function_lollipop_autoscale_maximum
+        }
+    }]
+
   }
 }
 
-resource "azurerm_resource_group" "lollipop_rg" {
-  count    = var.lollipop_enabled ? 1 : 0
-  name     = format("%s-lollipop-rg", local.common_project)
-  location = var.location
+resource "azurerm_resource_group" "lollipop_rg_itn" {
+  name     = format("%s-lollipop-rg-01", local.common_project_itn)
+  location = local.itn_location
 
   tags = var.tags
 }
 
+
+
 # Subnet to host admin function
-module "lollipop_snet" {
-  count                                     = var.lollipop_enabled ? 1 : 0
-  source                                    = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v4.1.15"
-  name                                      = format("%s-lollipop-snet", local.common_project)
-  address_prefixes                          = var.cidr_subnet_fnlollipop
-  resource_group_name                       = data.azurerm_virtual_network.vnet_common.resource_group_name
-  virtual_network_name                      = data.azurerm_virtual_network.vnet_common.name
-  private_endpoint_network_policies_enabled = false
+module "lollipop_snet_itn" {
```

Comment:
Are you destroying the existing instance?

P.S.: I suggest to avoid this module as it is a wrapper as the resource code would be:

```hcl
resource "azurerm_subnet" "lollipop_fn" {
  name                               = format("%s-lollipop-snet-01", local.common_project_itn)
  resource_group_name  = data.azurerm_virtual_network.common_vnet_italy_north.resource_group_name
  virtual_network_name  = data.azurerm_virtual_network.common_vnet_italy_north.name
  address_prefixes           = var.cidr_subnet_fnlollipop_itn

  service_endpoints = [
    "Microsoft.Web",
    "Microsoft.AzureCosmosDB",
    "Microsoft.Storage",
  ]

  delegation = {
    name = "default"
    service_delegation = {
      name = "Microsoft.Web/serverFarms"
      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
    }
  }

  private_link_service_network_policies_enabled = true
  private_endpoint_network_policies_enabled      = true
}
```

Both of them have 22 rows, but the resource has no dependencies.

P.S.: The [DX function module](https://github.com/pagopa/dx/tree/main/infra/modules/azure_function_app) creates the subnet and the private endpoints for you

Reply:
Thanks, done in [bf52787](https://github.com/pagopa/io-infra/pull/1037/commits/bf52787b58caa5c3999ec8e57d0806cacbc37f37)